rs/tests/src/tecdsa/tschnorr_message_sizes_test.rs

```diff
@@ -0,0 +1,183 @@
+        .expect("Failed to install NNS canisters");
+}
+
+fn test_message_sizes(subnet: SubnetSnapshot, limit: usize, log: &Logger) {
+    let node = subnet.nodes().next().unwrap();
+    let agent = node.with_default_agent(|agent| async move { agent });
+    let msg_can = block_on(MessageCanister::new(&agent, node.effective_canister_id()));
+    let cycles = SCHNORR_SIGNATURE_FEE;
+
+    for key_id in &make_schnorr_key_ids_for_all_algorithms() {
+        info!(log, "Getting the public key for {}", key_id);
+        let public_key = block_on(async {
+            get_public_key_with_logger(key_id, &msg_can, log)
+                .await
+                .unwrap()
+        });
+
+        let empty_message = vec![];
+        info!(log, "Getting signature of empty message for {}", key_id);
+        let signature = block_on(async {
+            get_signature_with_logger(empty_message.clone(), cycles, key_id, &msg_can, log)
+                .await
+                .unwrap()
+        });
+        info!(log, "Verifying signature of empty message for {}", key_id);
+        verify_signature(key_id, &empty_message, &public_key, &signature);
+
+        // Subtract 1 KIB to account for message overhead in addition to payload
+        let max_message = vec![0xabu8; limit - KIB];
+        info!(
+            log,
+            "Getting signature of message with size {} for {}",
+            max_message.len(),
+            key_id
+        );
+        let signature = block_on(async {
+            get_signature_with_logger(max_message.clone(), cycles, key_id, &msg_can, log)
+                .await
+                .unwrap()
+        });
+        info!(
+            log,
+            "Verifying signature of message with size {} for {}",
+            max_message.len(),
+            key_id
+        );
+        verify_signature(key_id, &max_message, &public_key, &signature);
+
+        let exceeding_message = vec![0xabu8; limit];
+        info!(
+            log,
+            "Getting signature of message with size {} for {}",
+            exceeding_message.len(),
+            key_id
+        );
+        let result = block_on(get_signature_with_logger(
+            exceeding_message,
+            cycles,
+            key_id,
+            &msg_can,
+            log,
+        ));
+        assert!(result.is_err());
+        info!(
+            log,
+            "Signature request failed successfully with: {:?}", result
+        )
+    }
+}
+
+pub fn test(env: TestEnv) {
+    let log = env.logger();
+    let topology_snapshot = env.topology_snapshot();
+    let mut subnets = topology_snapshot.subnets().skip(1);
+    let (schnorr_subnet, app_subnet) = (subnets.next().unwrap(), subnets.next().unwrap());
+
+    info!(log, "Testing tSchnorr message sizes using XNet");
+    test_message_sizes(app_subnet, XNET_LIMIT, &log);
+    info!(log, "Testing tSchnorr message sizes using local subnet");
+    test_message_sizes(schnorr_subnet, LOCAL_LIMIT, &log);
+}
```

Comment:
nit: can we split this into two functions:
```
pub fn test_app_subnet(env: TestEnv) {
    (...)
    test_message_sizes(app_subnet, XNET_LIMIT, &log);
}

pub fn test_schnorr_subnet(env: TestEnv) {
    (...)
    test_message_sizes(shnorr_subnet, XNET_LIMIT, &log);
}
```

And then in `rs/tests/consensus/tecdsa/tschnorr_message_sizes_test.rs`:
```
SystemTestGroup::new()
        .with_setup(tecdsa::tschnorr_message_sizes_test::config)
        .add_test(systest!(tecdsa::tschnorr_message_sizes_test::test_app_subnet))
        .add_test(systest!(tecdsa::tschnorr_message_sizes_test::test_schnorr_subnet))
        .execute_from_args()?;
```
This way, if I understand correctly, the test runner will treat them as two separate test cases but still will run the setup only once

Reply:
Nice idea, done